### PR TITLE
more features to Ryven flow integration

### PR DESCRIPTION
**Pull request by Licini**
Added various updates:
- Global `Run All` , `Enable Auto Update`, `Disable Auto Update` function
- More comprehensive Data inputs widgets (String, Choice)
- Nodes automatically turn red after code execution fail

image
image

### CODE REVIEW

**Comment by brgcode** [comment]
cool!

File: scripts/v120_flow.py [hunk 1]
```diff
@@ -10,26 +11,26 @@
 
 # Wrapping the function to ryven nodes
 @Node(viewer)
-def load_bunny() -> Mesh:
-    return Mesh.from_ply(compas.get('bunny.ply'))
+def load_mesh(path: str) -> Mesh:
+    return Mesh.from_ply(compas.get(path))
```
Comment (brgcode) [inline scripts/v120_flow.py:15]: we should add a `from_file` method that just checks the extension and then dispatches to the more specific methods...

File: src/compas_view2/flow/values.py [hunk 2]
```diff
@@ -8,32 +8,85 @@ class Value(rc.Node):
...
+    def view_place_event(self):
         self.update()
 
     def update_event(self, inp=-1):
         self.set_output_val(0, self.input(0))
 
 
-class Integer(Value):
+def Integer(default: int = 1, bounds: tuple = None):
+
```
Comment (brgcode) [inline src/compas_view2/flow/values.py:28]: docs

File: src/compas_view2/flow/values.py [hunk 3]
```diff
@@ -8,32 +8,85 @@ class Value(rc.Node):
...
+                dtype=rc.dtypes.Integer(default=default, bounds=bounds),
+                label='Int'
+            ),
+        ]
+
+    return IntegerNode
+
+
+def Float(default: float = 0.0, bounds: tuple = None):
+
```
Comment (brgcode) [inline src/compas_view2/flow/values.py:46]: docs

File: src/compas_view2/flow/values.py [hunk 4]
```diff
@@ -8,32 +8,85 @@ class Value(rc.Node):
...
+                dtype=rc.dtypes.Float(default=default, bounds=bounds),
+                label='Float'
+            ),
+        ]
+
+    return FloatNode
+
+
+def Choice(items: list, default=None):
+
```
Comment (brgcode) [inline src/compas_view2/flow/values.py:64]: docs

**Review by brgcode — APPROVED** [review]
LGTM.
just don't forget about the docs.
otherwise they never get filled in ... :)

**Comment by brgcode** [comment]
after merging this, could you add an example flow for CSG modeling?

File: scripts/v120_flow.py (hunk 1 above)
Comment (Licini) [reply scripts/v120_flow.py:15]: > LGTM. just don't forget about the docs. otherwise they never get filled in ... :)

haha sure doing that now

**Comment by Licini** [comment]
@brgcode Added docs to everything and some small clean up. Gonna just merge now and work on a csg example

**Pull request merged**